service/lib/agama/software/callbacks/base.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2025] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+
+module Agama
+  module Software
+    module Callbacks
+      # Base class for libzypp callbacks for sharing some common texts.
+      class Base
+        include Yast::I18n
+
+        # Constructor
+        def initialize
+          textdomain "agama"
+        end
+
+        # separator between multiple messages
+        def separator
+          "\n\n"
+        end
+
+        # label for the "retry" action
+        def retry_label
+          # TRANSLATORS: button label, try downloading the failed package again
+          _("Try again")
+        end
+
+        # label for the "continue" action
+        def continue_label
+          # TRANSLATORS: button label, ignore the failed download, skip package installation
+          _("Continue anyway")
+        end
+
+        # label for the "abort" action
+        def abort_label
+          # TRANSLATORS: button label, abort the installation completely after an error
+          _("Abort installation")
+        end
+
+        # dialog title
+        def title
+          # TRANSLATORS: Dialog title
+          _("Package installation failed")
+        end
```

Comment:
I actually meant a title of the dialog. Here you have added the title to the question, which is ok, but then you have had to include it as part of the question text and use CSS workarounds for having separated lines.